test/tests/executioners/eigen_convergence/tests

```diff
@@ -0,0 +1,44 @@
+[Tests]
+  design = 'EigenProblem.md'
+  issues = '#20095'
+  [testb]
+    type = CSVDiff
+    csvdiff = 'b_out.csv'
+    input = 'b.i'
+    expect_out = '4 Nonlinear'
+    absent_out = '5 Nonlinear'
+    requirement = 'The system shall be able to solve an eigenvalue problem using something other than the L2 norm of Bx for normalization and a native eigenvalue solver.'
+    rel_err = 3e-5
+  []
+  [test11]
+    type = CSVDiff
+    csvdiff = 'a_out_eigen_0001.csv'
+    input = 'a.i'
+    expect_out = '1[01] Nonlinear'
+    absent_out = '12 Nonlinear'
+    cli_args = '-eps_tol 2e-10'
+    requirement = 'The system shall be able to solve an eigenvalue problem using the L2 norm of Bx and the sign of its first nonzero entry for normalization with a SLEPc eigenvalue solver.'
+  []
+  [test7]
+    type = CSVDiff
+    csvdiff = 'a_out_eigen_0001.csv'
+    prereq = test11
+    input = 'a.i'
+    expect_out = '7 Nonlinear'
+    absent_out = '8 Nonlinear'
+    cli_args = 'Problem/bx_norm=fluxintegral -eps_tol 2e-10'
+    petsc_version = '>=3.20.0'
+    requirement = 'The system shall be able to solve an eigenvalue problem using the sign of the first nonzero entry of Bx combined with something other than the L2 norm of Bx for normalization with a SLEPc eigenvalue solver.'
+  []
+  [test4]
+    type = CSVDiff
+    csvdiff = 'a_out_eigen_0001.csv'
+    prereq = test7
+    input = 'a.i'
+    expect_out = '4 Nonlinear'
+    absent_out = '5 Nonlinear'
+    cli_args = 'Problem/bx_norm=fluxintegral -eps_power_sign_normalization 0 -eps_tol 2e-10'
+    petsc_version = '>=3.20.0'
+    requirement = 'The system shall be able to solve an eigenvalue problem using something other than the L2 norm of Bx for normalization and a SLEPc eigenvalue solver.'
```

Comment:
I noticed that this test takes more linear iterations per newton than `b.i`, any explanation or fix?

Reply:
we've talked about this in the past. In the SLEPc formulation we just use A to construct the preconditioner but in your old formulation (at least in this test) you include information from B as well. This means that the legacy preconditioning matrix more accurately reflects the Jacobian but you have told me this often leads to the matrix being more poorly conditioned

Reply:
Oh. Then this is completely fine because assembling B in real problems can be costly.

It seems that I cannot resolve a conversation, otherwise I can mark some as resolved.